### PR TITLE
Add support for restart policy update

### PR DESCRIPTION
**Pull request by mferon**
Hi all,

Here is a patch to be able to update the restart policy of a container. (Issue #1106)

A PR already exists (#1158), but it seems abandoned...

I'm open to every suggestion ;)

### CODE REVIEW

File: docker/api/container.py [hunk 1]
```diff
@@ -416,12 +416,13 @@ def unpause(self, container):
         res = self._post(url)
         self._raise_for_status(res)
 
-    @utils.minimum_version('1.22')
+    @utils.minimum_version('1.23')
```
Comment (shin-) [inline docker/api/container.py]: This should not be changed - people still need to be able to update containers on 1.22

File: docker/api/container.py [hunk 2]
```diff
@@ -445,6 +446,8 @@ def update_container(
             data['MemorySwap'] = utils.parse_bytes(memswap_limit)
         if kernel_memory:
             data['KernelMemory'] = utils.parse_bytes(kernel_memory)
+        if restart_policy:
```
Comment (shin-) [inline docker/api/container.py]: There needs to be a `version_lt` check here for 1.23 instead

**Comment by mferon** [comment]
Thank you for your review ;)

I updated the branch to fix what you said.

**Comment by shin-** [comment]
Thanks - LGTM!

**Comment by dnephin** [comment]
LGTM

**Pull request merged**